src/edu/washington/escience/myria/expression/CounterExpression.java

```diff
@@ -0,0 +1,76 @@
+package edu.washington.escience.myria.expression;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.expression.evaluate.ExpressionOperatorParameter;
+
+/**
+ * Return a sequence of numbers from 0 (inclusive) to operand (exclusive).
+ */
+public class CounterExpression extends UnaryExpression {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * This is not really unused, it's used automagically by Jackson deserialization.
+   */
+  @SuppressWarnings("unused")
+  private CounterExpression() {
+    super();
+  }
+
+  /**
+   * Takes the upper bound of the sequence to be returned.
+   *
+   * @param operand an expression that evaluates to a positive integer
+   */
+  public CounterExpression(final ExpressionOperator operand) {
+    super(operand);
+  }
+
+  @Override
+  public Type getOutputType(final ExpressionOperatorParameter parameters) {
+    Type operandType = getOperand().getOutputType(parameters);
```

Comment:
We can add a `checkOperandType` in `UnaryExpression` and call it as we did for `BinaryExpression` and `SplitExpression`.